buffer/src/main/java/io/netty/buffer/ReadableObject.java

```diff
@@ -0,0 +1,166 @@
+     *            less than {@code 0} or
+     *            greater than {@link #readerLimit()}.
+     */
+    ReadableObject readerPosition(long readerPosition);
+
+    /**
+     * Gets the read position limit (exclusive). The value of {@code readerPosition} must always
+     * be less than or equal to this limit.
+     */
+    long readerLimit();
+
+    /**
+     * Returns {@code true} if and only if {@link #readableBytes()} > {@code 0}.
+     */
+    boolean isReadable();
+
+    /**
+     * Returns {@code true} if and only if this buffer contains equal to or more than the specified number of elements.
+     */
+    boolean isReadable(int size);
+
+    /**
+     * Returns the number of readable bytes in this object.
+     */
+    long readableBytes();
+
+    /**
+     * Increases the current {@code readerPosition} by the specified
+     * {@code length} in this buffer.
+     *
+     * @throws IndexOutOfBoundsException
+     *         if {@code length} is greater than {@link #readableBytes()}
+     */
+    ReadableObject skipBytes(long length);
+
+    /**
+     * Returns a slice of this object's readable region. This method is
+     * identical to {@code r.slice(r.readerPosition(), r.readableBytes())}.
+     * This method does not modify {@code readerPosition}.
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the
+     * reference count will NOT be increased.
+     */
+    ReadableObject slice();
+
+    /**
+     * Returns a slice of this object's sub-region. This method does not modify
+     * {@code readerPosition}.
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the
+     * reference count will NOT be increased.
+     *
+     * @param position the starting position for the slice.
+     *
+     * @param length the size of the new slice relative to {@code position}.
+     *
+     * @throws IndexOutOfBoundsException
+     *         if any part of the requested region falls outside of the currently readable region.
+     */
+    ReadableObject slice(long position, long length);
+
+    /**
+     * Returns a new slice of this object's sub-region starting at the current
+     * {@link #readerPosition()} and increases the {@code readerPosition} by the size
+     * of the new slice (= {@code length}).
+     * <p>
+     * Also be aware that this method will NOT call {@link #retain()} and so the
+     * reference count will NOT be increased.
+     *
+     * @param length the size of the new slice
+     *
+     * @return the newly created slice
+     *
+     * @throws IndexOutOfBoundsException
+     *         if {@code length} is greater than {@link #readableBytes()}
+     */
+    ReadableObject readSlice(long length);
+
+    /**
+     * Return the underlying {@link ReadableObject} instance if this is a wrapper around another
+     * object (e.g. a slice of another object).
+     *
+     * @return {@code null} if this is not a wrapper
```

Comment:
consider just combining this with the previous statement. Looks strange by itself.